rsp/xxx_rest_api_design.md

```diff
@@ -0,0 +1,261 @@
+
+Parameters:
+* `type`: Restricts search to the specified type ("monster" or "player")
+
+### Returns
+
+```javascript
+[
+{
+    "id": 4,
+    "type": {
+        "player": {
+            "id": 1,
+            "name": "Vaelden",
+            "gold": 42,
+            "experience": 43,
+            "guild": "God"
+        }
+    },
+    "skin": 1,
+    "current_pv": 100,
+    "position": {
+        "x": 0.1,
+        "y": 42.42,
+        "map": 0,
+        "instance": 2
+    },
+    "stats": {
+        "level": 1,
+        "strength": 2,
+        "dexterity": 3,
+        "constitution": 4,
+        "intelligence": 5,
+        "precision": 6,
+        "wisdom": 7
+    }
+},
+{
+    "id": 5,
+    "type": {
+        "monster": {
+            "id": 3,
+            "name": "zombie",
+            "behaviour_tree": "zombie.bt"
+        }
+    },
+    "skin": 0,
+    "current_pv": 100,
+    "position": {
+        "x": 0.0,
+        "y": 10.0,
+        "map": 0,
+        "instance": 3
+    },
+    "stats": {
+        "level": 7,
+        "strength": 6,
+        "dexterity": 5,
+        "constitution": 4,
+        "intelligence": 3,
+        "precision": 2,
+        "wisdom": 1
+    }
+}
+]
+```
+
+## POST /instances/:id/spawn
+
+Spawn a specific mob in the instance
+
+### Body
+
+```javascript
+{
+    "monster_class": 3,
+    "x": 0.0,
+    "y": 10.0
+}
+```
+
+Optional parameters:
+* `x`: Default value 0.0
+* `y`: Default value 0.0
+
+### Returns
+
+```javascript
+{
+    "id": 6,
+    "type": {
+        "monster": {
+            "id": 3,
+            "name": "zombie",
+            "behaviour_tree": "zombie.bt"
+        }
+    },
+    "current_pv": 100,
+    "position": {
+        "x": 0.0,
+        "y": 10.0,
+        "map": 0,
+        "instance": 3
+    },
+    "stats": {
+        "level": 7,
+        "strength": 6,
+        "dexterity": 5,
+        "constitution": 4,
+        "intelligence": 3,
+        "precision": 2,
+        "wisdom": 1
+    }
+}
+```
+
+## DELETE /instances/:id/entities/:id
+
+Removes a mob from the game
+
+XXX: Should it kick players as well?
+
+## GET /players
+
+Get list of connected players
+
+### Returns
+
+```javascript
+[
+{
+    "id": 4,
+    "type": {
+        "player": {
+            "id": 1,
+            "name": "Vaelden",
+            "gold": 42,
+            "experience": 43,
+            "guild": "God"
+        }
+    },
+    "skin": 1,
+    "current_pv": 100,
+    "position": {
+        "x": 0.1,
+        "y": 42.42,
+        "map": 0,
+        "instance": 2
+    },
+    "stats": {
+        "level": 1,
+        "strength": 2,
+        "dexterity": 3,
+        "constitution": 4,
+        "intelligence": 5,
+        "precision": 6,
+        "wisdom": 7
+    }
+}
+]
+```
+## DELETE /players/:id
+
+Kicks a player from the game
+
+XXX: Banning - Should we do it from the management console (i.e. block account, then kick from
+                                                            Lycan?)
+## POST /shutdown
+
```

Comment:
>    If no one is going to use it because it's unstable what's the point of having it?
>    If everyone use it, then it's more or less de facto stable, so can we really change it?

I think you are a bit extreme, it won't be "everyone" or "no one". The problem is that we will need the functionality to be present _before_ we can guarantee that it is stable.

For example, I will most likely use those APIs if I start to write tests. I will need to spawn monsters, get their states etc. before we have fully defined what a monster is, what fields it includes ...

So I prefer having some documentation saying what is available for use / planned, rather than no documentation because we cannot guarantee its stability.

> What's the problem with removing stuff that you don't want to discuss and that you can avoid to have to deal with at all?

I don't want to avoid discussing it, I am just telling that the full implementation will not appear any time soon.